api/v1alpha1/blockdevice_types.go

```diff
@@ -254,7 +228,27 @@ const (
 	BlockDeviceUnknown BlockDeviceState = "Unknown"
 )
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
```

Comment:
Is a subresource required?

Reply:
Yes. Basically, the desired state is expressed in `spec` and observed state is present in `status` sub resource . Client-go libraries can make use of this for updates, etc.